generator/src/main/java/com/scottlogic/deg/generator/restrictions/StringRestrictions.java

```diff
@@ -1,28 +1,78 @@
 package com.scottlogic.deg.generator.restrictions;
 
-import com.scottlogic.deg.generator.constraints.StringConstraintsCollection;
-import com.scottlogic.deg.generator.generation.StringGenerator;
-import com.scottlogic.deg.generator.generation.IsinStringGenerator;
-import com.scottlogic.deg.generator.generation.RegexStringGenerator;
-import com.scottlogic.deg.generator.generation.SedolStringGenerator;
+import com.scottlogic.deg.generator.constraints.atomic.*;
+import com.scottlogic.deg.generator.generation.*;
+import com.scottlogic.deg.generator.walker.reductive.Merged;
 
-import java.util.Objects;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * https://github.com/ScottLogic/datahelix/blob/ws/experimental-data-constraint-solver/data-constraint-poc/src/main/java/com/scottlogic/deg/constrainer/util/RegexProcessor.java
  * https://github.com/ScottLogic/datahelix/blob/ws/experimental-data-constraint-solver/data-constraint-poc/src/main/java/com/scottlogic/deg/constrainer/RegexFieldConstraint.java#L133
  */
 public class StringRestrictions {
-    private final StringConstraintsCollection constraints;
-    public StringGenerator stringGenerator;
+    private static final int DEFAULT_SHORTER_THAN = 255 + 1;
+    private static final int MAX_SHORTER_THAN = 1000 + 1;
 
-    public StringRestrictions(StringConstraintsCollection constraints) {
+    final Set<AtomicConstraintState> constraints;
```

Comment:
This is a fairly large request, but i think this class would be a lot cleaner if we didn't store any atomicConstraints in this restrictions class.

if we could convert them at an early point to just integers that represent length options (shorter, longer, equal, not equal) lots of the complexity in this class would drop away

can convert before they come into the fieldspec, or in here as a first step